api.js

```diff
@@ -141,16 +162,23 @@ Api.prototype._prefixTitle = function (file) {
 	return prefix;
 };
 
-Api.prototype.run = function () {
+Api.prototype.run = function (files) {
 	var self = this;
 
-	return handlePaths(this.files)
+	this._reset();
+	this.explicitTitles = Boolean(files);
+	return handlePaths(files || this.files, this.excludePatterns)
```

Comment:
Should the `files` parameter just be required? Move the `handlePaths` logic outside the Api entirely.

The Api is not documented or to be considered stable, so there's no worry about a breaking change here.